src/app/data-planes/views/DataPlaneListView.vue

```diff
@@ -114,13 +114,8 @@ async function loadData(offset: number, dppParams: DataPlaneOverviewParameters =
   try {
     const { items, next } = await kumaApi.getAllDataplaneOverviewsFromMesh({ mesh }, params)
 
-    if (Array.isArray(items) && items.length > 0) {
-      dataPlaneOverviews.value = items
-      nextUrl.value = next
-    } else {
-      dataPlaneOverviews.value = []
-      nextUrl.value = null
-    }
+    nextUrl.value = next
+    dataPlaneOverviews.value = items ?? []
```

Comment:
Nice little simplification. Just out of interest, apart from when the HTTP request throws an error, is there any eventuality where items isn't an array?

Reply:
As far as I can tell, in general, our APIs prefer omitting a field when it’s empty (e.g. a field would be missing instead of being `[]`).

However, it seems `items` as returned by API endpoints for collection resources, when empty, is actually provided as `null` instead of `[]` or being omitted. Since we always use `items ?? []`, we would be fine in the case of `null` being provided _or_ the field being omitted. I wonder if we should mark the field as optional, too, in the event that it is actually omitted, but I’m not aware of that actually being possible.

(Personally I would prefer empty lists to be `[]` always and never `null` or omitted, but that’s not what we do)

Reply:
Yeah, it would be a good idea to make this maybe a bit chaotic even so we catch missed edge cases more easily (e.g. allow them to be `omitted`, too, so a check explicitly against `!== null` would be incorrect).

Reply:
Looks like at least this endpoint returns an empty array, so if all of the endpoints are the same (🤞 I would assume so) I think we are good for the mocks at least, actually I'll check in with backend offline to confirm

![Screenshot 2023-04-28 at 09 53 37](https://user-images.githubusercontent.com/554604/235102728-1db50e51-9916-4ea5-89da-1f70c28d067a.png)

Reply:
> However, it seems items as returned by API endpoints for collection resources, when empty, is actually provided as null instead of [] or being omitted.

From my checking and confirming offline with the backend peeps it seems like that is not the case ^. An empty `items` property will always return `[]` (as opposed to `null` or omission), and if it doesn't its a bug.

Could you double check whatever it was that made you think different for me just in case I'm missing something?

Reply:
At least `meshes/default/service-insights` seems to return `null` for `items`. I’m not changing this for the time being even though it would be nice to get rid of the `| null` type. Right now, this is the safer bet.

Reply:
Yeah for sure, makes sense 👍 . Glad this came up though. Did you want to make an issue for `meshes/default/service-insights` in kuma and link it here?

Reply:
Issue here: https://github.com/kumahq/kuma/issues/6639